unit_tests/RecordUnitTests.d

```diff
@@ -524,3 +524,25 @@ D");
 
 }
 
+@("issue #foo - translate record with name from macro")
+unittest {
+    assertTranslates(
+    q{
+
+#define __FSID_T_TYPE struct { int __val[2]; }
+typedef  __FSID_T_TYPE __fsid_t;
+typedef __fsid_t fsid_t;
+
+},
+    q{extern (C):
```

Comment:
I can give it some though if we can replace `q"D` with `q{}`, but `q"C` should stay. _If_ I want to change it would be for all code and in a separate commit/PR.